### PR TITLE
refactor: use clap as a commands parser

### PR DESCRIPTION
**Pull request by therustmonk**
Description
---
The PR refactors replace our own parser with a derived by the `clap` crate.
It also adds the `watch` command, and avoids unexpected status printing. 

More changes:
- integration tests use `non-interactive` mode
- added `--watch` parameter to the `non-interactive` mode to run a command with an interval
- interactive mode asks for confirmation on exit (to avoid accidental node termination)
- added the `watch` command to the interactive mode

Motivation and Context
---
Improve the user experience of the node's shell.
Simplify maintenance and adding new commands.
Also periodically status prnting breaks the rustyline and prompt chars not printed properly.

How Has This Been Tested?
---
Manually

### CODE REVIEW

File: applications/tari_base_node/src/commands/command/ban_peer.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+use crate::LOG_TARGET;
+
+/// Bans a peer
+#[derive(Debug, Parser)]
+pub struct ArgsBan {
+    /// hex public key or emoji id
+    node_id: UniNodeId,
+    /// length of time to ban the peer for in seconds
+    #[clap(default_value_t = std::u64::MAX)]
+    lenght: u64,
```
Comment (sdbondi) [inline applications/tari_base_node/src/commands/command/ban_peer.rs]:
```suggestion
    length: u64,
```
Maybe `time_secs` is better but up to you

**Review by sdbondi — COMMENTED** [review]
Looks great! Will wait for you to finish before approving

**Review by sdbondi — DISMISSED** [review]
Tested a few commands and working well 👍

**Comment by delta1** [comment]
bors retry

**Comment by delta1** [comment]
Seems like something has broken in the integration tests? @DenisKolodin

**Review by stringhandler — DISMISSED** [review]

**Review by stringhandler — APPROVED** [review]

**Pull request merged**